### ISSUE TITLE
Gmail search syntax for Remove-GSGmailMessage

### ISSUE THREAD

**Issue by jboiselle** [scrthq/PSGSuite#93]
Hi Nate,

I'm looking for some additional functionality that is closely related to the `Remove-GSGmailMessage` cmdlet. Specifically, feature parity with the [Gmail search syntax](https://support.google.com/mail/answer/7190?hl=en) functionality that a similar command in GAM currently has. I'd like to request one (or both) of the following features:

1. The ability to use `Remove-GSGmailMessage` with the Gmail search syntax. For example, if I wanted to delete all messages from a particular sender (`from:example@domain.com`).

2. A new cmdlet with the ability to modify labels on a particular email (also using the Gmail search syntax). For example, if I wanted to mark all messages from a particular Gmail search syntax query as spam or trash.

You can read the documentation for the GAM commands here:
https://github.com/jay0lee/GAM/wiki/ExamplesEmailSettings#modifying-user-emails

The reason I'm looking for this functionality is that I often use these commands to combat spam or phishing. Attackers will often send several emails with different message IDs. The search syntax would allow far greater flexibility in targeting these emails. I still find myself using GAM for removing emails specifically for the ability to use Gmail search syntax.

Thoughts?

Thanks,
Josh

**Comment by scrthq** [comment]
Hey Josh - Definitely good ideas! I use PSGSuite myself for almost the same purpose (among many other), so I completely get the goal here!

For item 1 - I'll check out expanding the functionality so it's wrapped into a single command, but you can currently use the pipeline though to do it in a short one-liner (see below). Would it be worth adding in an additional `MaxThreshold` parameter in case you're not doing something as precise as searching by MessageId and want to ensure you don't accidentally delete more than expected?

```powershell
# Trash everything not sent directly to me
Get-GSGmailMessageList -Filter "to:-me" | Remove-GSGmailMessage -Confirm:$false
```

![image](https://user-images.githubusercontent.com/12724445/46620189-aa690480-cae9-11e8-9e1d-eeafdd8d7e7a.png)

***

For item 2 - Definitely need to expand on label management! I'll work on this as I can, pull requests also very much accepted! 😄

**Comment by scrthq** [comment]
Alright, everything all fixed up and about to deploy in v2.15.2!

* Added in `Update-GSGmailMessageLabel` with help from @dwrusse and included `-Filter` parameter
* Updated `Remove-GSGmailMessage` with the same `-Filter` parameter
* Fixed the pipeline support so that `-User` isn't needed to pass through the pipeline so the pipeline works exactly as expected and pulls the `User` from the objects passed into it

Demo with WhatIf + Verbose showing new `Remove-GSGmailMessage` functionality:

![image](https://user-images.githubusercontent.com/12724445/46648449-c0fb7400-cb5a-11e8-80c1-b3b6cd147253.png)

**Comment by scrthq** [comment]
v2.15.2 has been deployed! Update and check it out, cheers!
